### PR TITLE
First draft at making the http.log use our own logging infrastucture.

### PR DESCRIPTION
**Pull request by chrisvest**
This currently fails a number of tests, because they still assume that they can configure the http.log through the logback xml file.
So while the actual logging part is more or less done, the configuration part is still up in the air.
Another thing that needs to be looked at, is the http log line format.
I've tried my best to make our logging framework print lines that look like the logback pattern, but I'm not sure if its good enough yet.

### CODE REVIEW

File: community/server/src/main/java/org/neo4j/server/AbstractNeoServer.java [hunk 1]
```diff
@@ -325,12 +324,16 @@ private void startWebServer() throws Exception
...
     {
-        if ( !httpLoggingProperlyConfigured() )
+        if ( !getConfig().get( http_logging_enabled ) )
         {
             return;
         }
 
-        webServer.setHttpLoggingConfiguration(config.get( http_log_config_file ), config.get( http_logging_enabled ));
+        System.out.println("HELLO THERE");
+        System.out.println(config.get( GraphDatabaseSettings.logs_directory ).toString() );
```
Comment (chrisvest) [inline community/server/src/main/java/org/neo4j/server/AbstractNeoServer.java]: Debugging left overs?

Comment (srbaker) [inline community/server/src/main/java/org/neo4j/server/AbstractNeoServer.java]: Hah, yes, thanks. Fixed.

Comment (benbc) [inline community/server/src/main/java/org/neo4j/server/AbstractNeoServer.java]: AHEM!

File: community/server/src/main/java/org/neo4j/server/AbstractNeoServer.java [hunk 2]
```diff
@@ -325,12 +324,16 @@ private void startWebServer() throws Exception
...
+
+        webServer.setHttpLoggingConfiguration(config.get( GraphDatabaseSettings.logs_directory ),
+                config.get( http_logging_enabled ) );
```
Comment (benbc) [inline community/server/src/main/java/org/neo4j/server/AbstractNeoServer.java]: The last parameter is always true. Can't we get rid of it?

File: community/server/src/main/java/org/neo4j/server/web/AsyncRequestLog.java [hunk 3]
```diff
@@ -0,0 +1,107 @@
...
+    private final Log log;
+    private final ExecutorService asyncLogProcessingExecutor;
+    private final AsyncEvents<AsyncLogEvent> asyncEventProcessor;
+
+    public AsyncRequestLog( FileSystemAbstraction fs, String logFile ) throws IOException
+    {
+        NamedThreadFactory threadFactory = new NamedThreadFactory( "HTTP-Log-Rotator", true );
+        ExecutorService rotationExecutor = Executors.newCachedThreadPool( threadFactory );
+        Supplier<OutputStream> outputSupplier = new RotatingFileOutputStreamSupplier(
+                fs, new File( logFile ), 0, TimeUnit.DAYS.toMillis( 1 ), 7, rotationExecutor );
```
Comment (benbc) [inline community/server/src/main/java/org/neo4j/server/web/AsyncRequestLog.java]: Have you intentionally left the rotation unconfigurable?

File: community/server/src/main/java/org/neo4j/server/AbstractNeoServer.java [hunk 4]
```diff
@@ -325,12 +324,13 @@ private void startWebServer() throws Exception
...
     {
-        if ( !httpLoggingProperlyConfigured() )
+        if ( !getConfig().get( http_logging_enabled ) )
         {
             return;
         }
 
-        webServer.setHttpLoggingConfiguration(config.get( http_log_config_file ), config.get( http_logging_enabled ));
+        webServer.setHttpLoggingConfiguration(config.get( GraphDatabaseSettings.logs_directory ),
+                config.get( http_logging_enabled ) );
```
Comment (benbc) [inline community/server/src/main/java/org/neo4j/server/AbstractNeoServer.java]: Second parameter is always true. You can just get rid of itl.

File: community/server/src/test/java/org/neo4j/server/web/logging/HTTPLoggingDocIT.java [hunk 5]
```diff
@@ -102,149 +86,12 @@ public void givenExplicitlyDisabledServerLoggingConfigurationShouldNotLogAccesse
...
-
-            // then
-            File httpLog = new File( logDirectory, "http.log" );
-            assertEventually( "request appears in log", fileContentSupplier( httpLog ), containsString( query ), 5, TimeUnit.SECONDS );
-        }
-        finally
-        {
-            server.stop();
-        }
-    }
```
Comment (benbc) [inline community/server/src/test/java/org/neo4j/server/web/logging/HTTPLoggingDocIT.java]: Do we not still want a test like this?

File: community/server/src/main/java/org/neo4j/server/web/AsyncRequestLog.java (hunk 3 above)
Comment (srbaker) [inline community/server/src/main/java/org/neo4j/server/web/AsyncRequestLog.java]: For now. Just haven't done it yet.

File: community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java [hunk 6]
```diff
@@ -173,13 +173,11 @@ private ThirdPartyJaxRsPackage createThirdPartyJaxRsPackage( String packageAndMo
     @Description("Enable HTTP request logging.")
     Setting<Boolean> http_logging_enabled = setting( "org.neo4j.server.http.log.enabled", BOOLEAN, FALSE );
```
Comment (benbc) [inline community/server/src/main/java/org/neo4j/server/configuration/ServerSettings.java]: Could you change this to `dbms.logs.http.enabled`, please?

File: community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java [hunk 7]
```diff
@@ -428,11 +411,8 @@ private int countSet( boolean... booleans )
         return count;
     }
 
-    private void loadRequestLogging()
+    private void loadRequestLogging() throws IOException
```
Comment (benbc) [inline community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java]: Does this really throw a new exception?

File: community/server/src/test/java/org/neo4j/server/integration/ServerConfigIT.java [hunk 8]
```diff
@@ -49,23 +49,6 @@
...
-                ServerSettings.http_log_config_file.name(), configValue )
-                .build();
-
-        // When
-        server.start();
-
-        // Then
-        ObjectName name = getObjectName( server.getDatabase().getGraph(), ConfigurationBean.CONFIGURATION_MBEAN_NAME );
-        assertThat( getAttribute( name, ServerSettings.http_log_config_file.name() ), equalTo( (Object)configValue ) );
-    }
```
Comment (benbc) [inline community/server/src/test/java/org/neo4j/server/integration/ServerConfigIT.java:66]: This test isn't an HTTP logging test. You need to reinstate it with a different setting.

File: community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java (hunk 7 above)
Comment (srbaker) [inline community/server/src/main/java/org/neo4j/server/web/Jetty9WebServer.java]: No.

**Pull request merged**